src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -176,6 +185,25 @@ void BindingManager::HandleDeviceConnectionFailure(PeerId peerId, CHIP_ERROR err
 {
     // Simply release the entry, the connection will be re-established as needed.
     ChipLogError(AppServer, "Failed to establish connection to node 0x" ChipLogFormatX64, ChipLogValueX64(peerId.GetNodeId()));
+
+    for (PendingNotificationEntry pendingNotification : mPendingNotificationMap)
+    {
+        EmberBindingTableEntry entry = BindingTable::GetInstance().GetAt(pendingNotification.mBindingEntryId);
+
+        PeerId peer = PeerIdForNode(mInitParams.mFabricTable, entry.fabricIndex, entry.nodeId);
+
+        if (peerId == peer)
+        {
+            BindingManagerContext * context = static_cast<BindingManagerContext *>(pendingNotification.mContext);
+
+            context->DecrementConsumersNumber();
```

Comment:
This is the part I don't get.  The `PendingNotificationEntry` still exists in `mPendingNotifications`, and is still pointing to `context`, right?  But this is explicitly removing the refcount that the pending notification was holding...  Don't we end up with a dangling pointer as a result?

Reply:
Ok, that's right, now I get what you mean. I must misunderstood something on the previous review. I think that this notification should not be kept, after handle will be called, so I added removing it here, similarly as it's done in `HandleDeviceConnected`